### ISSUE TITLE
Homebrew tap support

### ISSUE THREAD

**Issue by pat-s** [thegeeklab/git-sv#118]
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap - would make updating a bit more convenient than manually updating from the release assets.

**Comment by xoxys** [comment]
In general, I'm not against it, but as I'm not a Mac user, its hard for me to maintain it. What concerns me the most is https://github.com/Homebrew/brew/blob/master/docs/Taps.md#repository-naming-conventions-and-assumptions as I really don't want to maintain a second repo to support it.

**Comment by pat-s** [comment]
Yes, that's required. I think you can provide multiple formulas in one repo/tap.

Of course anyone could provide/create a tap and create a formula, but ideally (trust/security) the tap would be hosted within the same namespace that hosts the source as well (i.e. `thegeeklab` org).

Another possible formula would be `ansible-doctor` that could be provided there. HTH if needed...

**Comment by xoxys** [comment]
For me that's out of scope as I can't maintain it. If you would like to take care, I could create a repo and invite you as maintainer.

**Comment by xoxys** [comment]
Im also happy to link an external repo in the readme as community support to create some "trust".

**Comment by pat-s** [comment]
Fully understand that you don't wanna maintain it.
I think a repo in your namespace makes most sense, even if maintained externally. Also because n > 1 and you might create even more CLI tools in the future ;)

Another reason to host in your org might be that you can auto-bump the formula upon new cli releases by issueing `brew bump-formula-pr --strict --url=https://github.com/user/my-software/archive/refs/tags/${{ github.event.release.tag_name }}.tar.gz --tap=user/homebrew-tap` during a release event. And you don't wanna do this so some external repo usually.

tl;dr: happy to configure it, ideally maintenance should be close to 0 once set up.

**Comment by xoxys** [comment]
Cool, thanks! I have created the repo https://github.com/thegeeklab/homebrew-tap and invited you as maintainer. Please add a MIT license file, you can copy the one from this repo.

**Comment by pat-s** [comment]
The tap is now functional.

```sh
❯ brew reinstall git-sv
==> Fetching thegeeklab/tap/git-sv
==> Downloading https://github.com/thegeeklab/git-sv/archive/refs/tags/v1.0.5.tar.gz
Already downloaded: /Users/pjs/Library/Caches/Homebrew/downloads/436c7ee2c9d859f19cb166c5ed404762c7aab38c199e6079cd39d0787f520f2f--git-sv-1.0.5.tar.gz
==> Reinstalling thegeeklab/tap/git-sv
==> go build -ldflags=-s -w -X main.BuildVersion=v1.0.5 -X main.BuildDate=2024-10-24T22:15:58Z ./cmd/git-sv
🍺  /opt/homebrew/Cellar/git-sv/1.0.5: 6 files, 6.9MB, built in 1 second
==> Running `brew cleanup git-sv`...

which git-sv
/opt/homebrew/bin/git-sv

git-sv --version
git-sv version=v1.0.5 date=2024-10-24T22:15:58Z
```

For updates: `brew bump-formula-pr -v --no-browse --no-fork --strict --version <version> git-sv` should create a PR in the repo. This however relies on the following assumptions:

- the user executing it needs write permissions in the tap repo, so it can push to a branch
- the user's git config must be configured
- before running the update command, the formula must be installed so the URL/tap is know

So I'd envision it something along the lines of

```yaml
  - name: bump-homebrew-tap
    image: docker.io/homebrew/brew:4.4.2
    secrets: [github_token]
    commands:
      - # configure git config for bot user -> git config [...]
      - brew tap thegeeklab/tap
      - brew install git-sv
      - brew bump-formula-pr -v --no-browse --no-fork --strict --version ${{ github.event.release.tag_name }} git-sv
```

I can test this on the next upcoming release and try to perform the linked actions in a container locally and see if this leads to a PR. Afterwards these actions could be executed in your release pipeline from the bot user (assuming the bot user gets added to the repo with w perms).
If you also want to skip the manual merge of the PR, automerge could be set up in the tap repo including a dummy check that tests whether the author is the bot user (just an idea).

**Comment by xoxys** [comment]
Awesome, thanks a lot! Looks like renovate can take care of it https://docs.renovatebot.com/modules/manager/homebrew/ and I would prefer this way. 

However, we would need to programmatically extract the version in the test part https://github.com/thegeeklab/homebrew-tap/blob/main/Formula/git-sv.rb#L29 from e.g. the url metadata.

**Comment by pat-s** [comment]
Good idea, I wasn't aware that there is a manager even!

Seems like the manager doesn't need any custom config or rules: https://github.com/thegeeklab/homebrew-tap/issues/2

**Comment by xoxys** [comment]
Thanks again, I'm going to close this issue as completed.
